api/src/middleware/request_context.py

```diff
@@ -108,8 +111,8 @@ def is_user_email_restricted() -> bool:
     Check if an email's domain is restricted (e.g., for WIP visibility).
     """
     request_context = get_request_context()
-    if not isinstance(request_context, RequestContext):
```

Comment:
The request_context is actually a dictionary rather than a RequestContext class. FYI, the request_context is storage per request.